sys/include/net/ipv4/hdr.h

```diff
@@ -0,0 +1,383 @@
+     *
+     * This module provides helper functions to set, get, and check these
+     * fields accordingly:
+     * * ipv4_hdr_set_flags()
+     * * ipv4_hdr_get_flags()
+     * * ipv4_hdr_set_fo()
+     * * ipv4_hdr_get_fo()
+     */
+    network_uint16_t fl_fo;
+    uint8_t ttl;            /**< time to live for this packet */
+    uint8_t protocol;       /**< protocol of this packet */
+    network_uint16_t csum;  /**< checksum of this packet */
+    ipv4_addr_t src;        /**< source address of this packet */
+    ipv4_addr_t dst;        /**< destination address of this packet */
+} ipv4_hdr_t;
+
+
+/**
+ * @brief   Sets the version field of @p hdr to 6
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ */
+static inline void ipv4_hdr_set_version(ipv4_hdr_t *hdr)
+{
+    hdr->v_ih_ts.u8[0] &= 0x0f;
+    hdr->v_ih_ts.u8[0] |= 0x40;
+}
+
+/**
+ * @brief   Gets the value of the version field of @p hdr
+ *
+ * @param[in] hdr   Pointer to an IPv4 header.
+ *
+ * @return  Value of the version field of @p hdr.
+ */
+static inline uint8_t ipv4_hdr_get_version(ipv4_hdr_t *hdr)
+{
+    return ((hdr->v_ih_ts.u8[0]) >> 4);
+}
+
+/**
+ * @brief   Sets the Internet Header Length field of @p hdr
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ * @param[in] ihl  The new value of Internet Header Length.
+ */
+static inline void ipv4_hdr_set_ihl(ipv4_hdr_t *hdr, uint8_t ihl)
+{
+    hdr->v_ih_ts.u8[0] &= 0xf0;
+    hdr->v_ih_ts.u8[0] |= 0x0f & ihl;
+}
+
+/**
+ * brief Gets the value of the Internet Header Length field of @p hdr
+ *
+ * @param[in] hdr   Pointer to an IPv4 header.
+ *
+ * @return Value of the Internet Header Length field of @p hdr
+ */
+static inline uint8_t ipv4_hdr_get_ihl(ipv4_hdr_t *hdr)
+{
+    return (hdr->v_ih_ts.u8[0] & 0x0f);
+}
+
+/**
+ * @brief   Sets the Type of Service field of @p hdr
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ * @param[in] ts  The new value of Type of Service
+ */
+static inline void ipv4_hdr_set_ts(ipv4_hdr_t *hdr, uint8_t ts)
+{
+    hdr->v_ih_ts.u8[1] &= 0;
+    hdr->v_ih_ts.u8[1] |= ts;
+}
+
+/**
+ * brief Gets the value of the Type of Service field of @p hdr
+ *
+ * @param[in] hdr   Pointer to an IPv4 header.
+ *
+ * @return Value of the Type of Service field of @p hdr
+ */
+static inline uint8_t ipv4_hdr_get_ts(ipv4_hdr_t *hdr)
+{
+    return (hdr->v_ih_ts.u8[1]);
+}
+
+/**
+ * @brief   Sets the Total Length field of @p hdr
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ * @param[in] tl  The new value of Total Length
+ */
+static inline void ipv4_hdr_set_tl(ipv4_hdr_t *hdr, uint16_t tl)
+{
+    hdr->tl = byteorder_htons(tl);
+}
+
+/**
+ * brief Gets the value of the Total Length field of @p hdr
+ *
+ * @param[in] hdr   Pointer to an IPv4 header.
+ *
+ * @return Value of the Total Length field of @p hdr
+ */
+static inline uint16_t ipv4_hdr_get_tl(ipv4_hdr_t *hdr)
+{
+    return byteorder_ntohs(hdr->tl);
```

Comment:
Yup, it's not necessary. Just made it for consistence with other hdr files (e.g https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/ipv6/hdr.h#L217)